sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -57,6 +60,12 @@ public BufferHolder(UnsafeRow row, int initialSize) {
     this.row.pointTo(buffer, buffer.length);
   }
 
+  public BufferHolder(int initialSizeInBytes) {
```

Comment:
This is a special use of `BufferHolder`. Better to add few comments explaining it.